test/call.test.ts

```diff
@@ -22,6 +22,10 @@ describe("call", () => {
     )).resolves.toEqual(42);
   });
 
+  it("evaluates a promise directly with `until`", async () => {
+    await expect(run(() => until(Promise.resolve(42)))).resolves.toEqual(42);
+  });
+
```

Comment:
We should put this in its own file and check both the negative, positive, and halt cases.

Reply:
@cowboyd what would you expect to happen in the halt case?